src/util/prolog/prologStringManipulation.ts

```diff
@@ -1,41 +1,122 @@
-const fetchFileExtension = /(.*)\.[^\.]*/;
-const fetchModuleDeclaration = /:-( |)module\([^)]*\) *\.[\r\t\f\v ]*\n?/g;
-const fetchPartialLineCommentsRegExp = /(^[^\n%]+)%[^\n\r]*(\r?\n)/gm;
-const fetchFullLineCommentsRegExp = /^[\r\t\f\v ]*%[^\n]*\n/gm;
-const fetchEmptyLinesRegExp = /\n(?:\s)*\n/gm;
-const fetchDoubleWhitespacesRegExp = /  */gm;
+import {fetchAllMatchesForAGroup} from "../regExpUtils";
+import builtInPredicates from "./builtInPredicates";
 
 export function removeFileExtension(path: string) {
+	const fetchFileExtension = /(.*)\.[^\.]*/;
+
```

Comment:
Nit: Is there a reason for this empty line? You seem to have a similar style with the other functions. While some people might consider the code more readable this way, some also consider it more "disconnected", as in: They feel like after the empty line something conceptually different should happen. But in reality it's just another command inside the same function. I consider myself to be one of the latter people, but mostly it's just a choice of design and I would also be okay if you leave it this way.

Reply:
My thinking here was that it does something different. The first part is just a setup which didn't even used to be inside the function but at the beginning of the file. The actually process happens only after.